frontend/__tests__/components/utils/name-value-editor.spec.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+
+import { NameValueEditor } from '../../../public/components/utils/name-value-editor';
+import { DragDropContext } from 'react-dnd';
+import HTML5Backend from 'react-dnd-html5-backend';
+
+describe('NameValueEditor', () => {
```

Comment:
Similar to above comment, I might organizationally add a second `describe()` and explicitly name the feature being tested, in most case, a `propertyName` and a `value` combo:

```JavaScript
describe('NameValueEditor', () => {
   describe('When supplied with attributes nameString and valueString' => {
       it('renders header correctly', () => {
          // proof 
       });  
       // any other things it does do, or does NOT do
   });
   
   describe('When readonly attribute is "true"', () => {
      it('renders PairElement buttons', () => {
        // proof
      });
      // any other things it does do, or does NOT do
   });
   
   describe('When readonly attribute is "falose" and allowSorting is "true"', () => {
      it('renders PairElement buttons', () => {
        // proof
      });
      // any other things it does do, or does NOT do
   });
});
```
This might be more scalable.

Reply:
This is all set; please see the newest commit.